parsely/src/test/java/com/parsely/parselyandroid/LocalStorageRepositoryTest.kt

```diff
@@ -0,0 +1,123 @@
+package com.parsely.parselyandroid
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import java.io.File
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class LocalStorageRepositoryTest {
+
+    private lateinit var sut: LocalStorageRepository
+    private val context = ApplicationProvider.getApplicationContext<Context>()
+
+    @Before
+    fun setUp() {
+        sut = LocalStorageRepository(context)
+    }
+
+    @Test
+    fun `when expelling stored event, then assert that it has no effect`() {
+        // given
+        sut.persistQueue((1..100).map { mapOf("index" to it) })
+
+        // when
+        sut.expelStoredEvent()
+
+        // then
+        assertThat(sut.getStoredQueue()).hasSize(100)
+    }
+
+    @Test
+    fun `given the list of events, when persisting the list, then querying the list returns the same result`() {
+        // given
+        val eventsList = (1..10).map { mapOf("index" to it) }
+
+        // when
+        sut.persistQueue(eventsList)
+
+        // then
+        assertThat(sut.getStoredQueue()).hasSize(10).containsExactlyInAnyOrderElementsOf(eventsList)
+    }
+
+    @Test
+    fun `given no locally stored list, when requesting stored queue, then return an empty list`() {
+        assertThat(sut.getStoredQueue()).isEmpty()
+    }
+
+    @Test
+    fun `given stored queue with some elements, when persisting in-memory queue, then assert there'll be no duplicates and queues will be combined`() {
+        // given
+        val storedQueue = (1..5).map { mapOf("index" to it) }
+        val inMemoryQueue = (3..10).map { mapOf("index" to it) }
+        sut.persistQueue(storedQueue)
+
+        // when
+        sut.persistQueue(inMemoryQueue)
+
+        // then
+        val expectedQueue = (1..10).map { mapOf("index" to it) }
+        assertThat(sut.getStoredQueue()).hasSize(10).containsExactlyInAnyOrderElementsOf(expectedQueue)
+    }
+
+    @Test
+    fun `given stored queue, when purging stored queue, then assert queue is purged`() {
+        // given
+        val eventsList = (1..10).map { mapOf("index" to it) }
+        sut.persistQueue(eventsList)
+
+        // when
+        sut.purgeStoredQueue()
+
+        // then
+        assertThat(sut.getStoredQueue()).isEmpty()
+    }
+
+    @Test
+    fun `given stored file with serialized events, when querying the queue, then list has expected events`() {
```

Comment:
1. Praise (❤️): Kudos on creating this test, nicely done! 💯 
2. Question (❓): How could I or anyone else later-on verify the contents of this `parsely-events.ser` file? Maybe something to document somewhere, like how to (re)generate this file. 🤔

Reply:
Thanks! That was my worry, as I haven't seen this kind of idea for local storage (usually, it's more like JSON e.g.).

About regeneration: you'd have to write a script that would take the content of `valid-java-parsely-events.ser` file and deserialize to ArrayList: this is what this test does, though. I'd skip the documentation for now, as I don't see anyone would like to modify this file: it's more like a "database migration test".